packages/smart/selective.go

```diff
@@ -107,6 +136,11 @@ func (sc *SmartContract) selectiveLoggingAndUpd(fields []string, ivalues []inter
 				addSQLWhere += whereFields[i] + "= '" + escapeSingleQuotes(whereValues[i]) + "' AND "
 			}
 		}
+		if isKeyTable {
+			if isEcosystem := getParam(whereFields, `ecosystem`); isEcosystem < 0 {
+				addSQLWhere += fmt.Sprintf("ecosystem = '%s' AND ", keyEcosystem)
```

Comment:
possible sql injection, use bind variables

Reply:
I have changed this to %d and converter.StrToInt64(keyEcosystem)